bot/exts/filtering/_utils.py

```diff
@@ -0,0 +1,222 @@
+
+    return subclasses
+
+
+def clean_input(string: str) -> str:
+    """Remove zalgo and invisible characters from `string`."""
+    # For future consideration: remove characters in the Mc, Sk, and Lm categories too.
+    # Can be normalised with form C to merge char + combining char into a single char to avoid
+    # removing legit diacritics, but this would open up a way to bypass _filters.
+    no_zalgo = ZALGO_RE.sub("", string)
+    return INVISIBLE_RE.sub("", no_zalgo)
+
+
+def past_tense(word: str) -> str:
+    """Return the past tense form of the input word."""
+    if not word:
+        return word
+    if word.endswith("e"):
+        return word + "d"
+    if word.endswith("y") and len(word) > 1 and word[-2] not in "aeiou":
+        return word[:-1] + "ied"
+    return word + "ed"
+
+
+def to_serializable(item: Any) -> Union[bool, int, float, str, list, dict, None]:
+    """Convert the item into an object that can be converted to JSON."""
+    if isinstance(item, (bool, int, float, str, type(None))):
+        return item
+    if isinstance(item, dict):
+        result = {}
+        for key, value in item.items():
+            if not isinstance(key, (bool, int, float, str, type(None))):
+                key = str(key)
+            result[key] = to_serializable(value)
+        return result
+    if isinstance(item, Iterable):
+        return [to_serializable(subitem) for subitem in item]
+    return str(item)
+
+
+@cache
+def resolve_mention(mention: str) -> str:
+    """Return the appropriate formatting for the mention, be it a literal, a user ID, or a role ID."""
+    guild = bot.instance.get_guild(Guild.id)
+    if mention in ("here", "everyone"):
+        return f"@{mention}"
+    try:
+        mention = int(mention)  # It's an ID.
+    except ValueError:
+        pass
+    else:
+        if any(mention == role.id for role in guild.roles):
+            return f"<@&{mention}>"
+        else:
+            return f"<@{mention}>"
+
+    # It's a name
+    for role in guild.roles:
+        if role.name == mention:
+            return role.mention
+    for member in guild.members:
+        if str(member) == mention:
+            return member.mention
+    return mention
+
+
+def repr_equals(override: Any, default: Any) -> bool:
+    """Return whether the override and the default have the same representation."""
+    if override is None:  # It's not an override
+        return True
+
+    override_is_sequence = isinstance(override, (tuple, list, set))
+    default_is_sequence = isinstance(default, (tuple, list, set))
+    if override_is_sequence != default_is_sequence:  # One is a sequence and the other isn't.
+        return False
+    if override_is_sequence:
+        if len(override) != len(default):
+            return False
+        return all(str(item1) == str(item2) for item1, item2 in zip(set(override), set(default)))
+    return str(override) == str(default)
+
+
+def starting_value(type_: type[T]) -> T:
```

Comment:
We should probably avoid accessing `__args__` and `__origin__` directly and use `typing.get_args` and `typing.get_origin` instead, since these apis seem to almost always change between versions, https://canary.discord.com/channels/267624335836053506/891788761371906108/1071456037363789865. 

Checking `if get_origin(type_) in (typing.Union, types.UnionType):` seems it could work here?

Reply:
Good point

Reply:
333bbf6